mesonbuild/environment.py

```diff
@@ -274,7 +274,7 @@ def __init__(self, source_dir, build_dir, options):
             self.cross_info = CrossBuildInfo(self.coredata.cross_file)
         else:
             self.cross_info = None
-        self.cmd_line_options = options.cmd_line_options
+        self.cmd_line_options = copy.copy(options.cmd_line_options)
```

Comment:
If you don't need a deep copy, this can just be `options.cmd_line_options.copy()`?

Reply:
Ok, didn't know dict had a copy() method. Done.